skfda/exploratory/visualization/_baseplot.py

```diff
@@ -0,0 +1,82 @@
+"""BasePlot Module.
+
+This module contains the abstract class of which inherit all
+the visualization modules, containing the basic functionality
+common to all of them.
+"""
+
+from abc import ABC, abstractmethod
+from typing import List, Optional, Sequence, Union
+
+import matplotlib.pyplot as plt
+from matplotlib.artist import Artist
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
+
+from ._utils import _figure_to_svg
+
+
+class BasePlot(ABC):
+    """
+    BasePlot class.
+
+    Attributes:
+        artists: List of Artist objects corresponding
+            to every instance of our plot. They will be used to modify
+            the visualization with interactivity and widgets.
+        fig: Figure over with the graphs are plotted.
+        axes: Sequence of axes where the graphs are plotted.
+    """
+
+    @abstractmethod
+    def __init__(
+        self,
+    ) -> None:
+        self.artists: List[Artist] = []
```

Comment:
Shouldn't this be an array of length `self.n_samples`?

Reply:
I think you can create an array of fixed length.

Reply:
do you mean a numpy array? or just a normal one

Reply:
Numpy array.